src/wallet/wallet.cpp

```diff
@@ -1359,6 +1359,308 @@ void CWallet::SyncMetaData(pair<typename TxSpendMap<T>::iterator, typename TxSpe
     }
 }
 
+//
+// Zcash transaction output selectors
+//
+
+std::optional<ZTXOSelector> CWallet::ToZTXOSelector(const libzcash::PaymentAddress& addr, bool requireSpendingKey) const {
+    auto self = this;
+    std::optional<ZTXOPattern> pattern = std::nullopt;
+    std::visit(match {
+        [&](const CKeyID& addr) {
+            if (!requireSpendingKey || self->HaveKey(addr)) {
+                pattern = addr;
+            }
+        },
+        [&](const CScriptID& addr) {
+            if (!requireSpendingKey || self->HaveCScript(addr)) {
+                pattern = addr;
+            }
+        },
+        [&](const libzcash::SaplingPaymentAddress& addr) {
+            if (!requireSpendingKey || self->HaveSaplingSpendingKeyForAddress(addr)) {
+                pattern = addr;
+            }
+        },
+        [&](const libzcash::SproutPaymentAddress& addr) {
+            if (!requireSpendingKey || self->HaveSproutSpendingKey(addr)) {
+                pattern = addr;
+            }
+        },
+        [&](const libzcash::UnifiedAddress& ua) {
+            auto ufvkId = this->FindUnifiedFullViewingKey(ua);
+            if (ufvkId.has_value()) {
+                // TODO: at present, the `false` value for the `requireSpendingKey` argument
+                // is not respected for unified addresses, because we have no notion of
+                // an account for which we do not control the spending key. An alternate
+                // approach would be to use the UFVK directly in the case that we cannot
+                // determine a local account.
```

Comment:
I'm not sure what to recommend here because I don't know what the effects on visible behaviour would be.

Reply:
I think that essentially we should defer judgement on this; it might be that we get rid of `AccountZTXOPattern` in the future and simply return a UFVK instead, guarded by the check against the flag. Alternately, `AccountZTXOPattern` could become `UFVKZTXOPattern` and use a `UfvkId` value in the place where we currently have `AccountId`. Suffice it to say that this will work for now.